src/Symfony/Component/HttpClient/Response/TransportResponseTrait.php

```diff
@@ -37,12 +39,12 @@ trait TransportResponseTrait
 
     /** @var object|resource */
     private $handle;
-    private $id;
-    private $timeout = 0;
-    private $inflate;
-    private $finalInfo;
-    private $canary;
-    private $logger;
+    private int|string $id;
+    private ?float $timeout = 0;
+    private \InflateContext|bool|null $inflate = null;
+    private ?array $finalInfo = null;
+    private ?Canary $canary = null;
```

Comment:
canary shouldn't be nullable IMO. otherwise close method makes an invalid call to the nullable property

Reply:
We could. But that property is not used in `MockResponse` (which overrides the `close()` method anyway).

Reply:
I've removed the nullability.